File: python/get_person_kellokortti.py
import os
import requests
from requests.auth import HTTPBasicAuth


def get_person(person_id: str):
    """Fetch a person from the Kellokortti API using Basic authentication."""
    api_url = os.environ.get("KK_API_URL", "https://api.kellokortti.fi/api")
    username = os.environ.get("KK_USERNAME")
    password = os.environ.get("KK_PASSWORD")
    if not username or not password:
        raise ValueError("KK_USERNAME and KK_PASSWORD must be set")

    url = f"{api_url}/api/v1/person/{person_id}"
    response = requests.get(url, auth=HTTPBasicAuth(username, password))
    response.raise_for_status()
    return response.json()


if __name__ == "__main__":
    pid = os.environ.get("KK_PERSON_ID", "testuser")
    print(get_person(pid))

File: php/get_person_kellokortti.php
<?php
function get_kellokortti_person($person_id) {
    $apiUrl = getenv('KK_API_URL') ?: 'https://api.kellokortti.fi/api';
    $username = getenv('KK_USERNAME');
    $password = getenv('KK_PASSWORD');
    if (!$username || !$password) {
        throw new Exception('KK_USERNAME and KK_PASSWORD must be set');
    }
    $url = "$apiUrl/api/v1/person/$person_id";
    $opts = array(
        'http' => array(
            'method' => 'GET',
            'header' => 'Authorization: Basic ' . base64_encode("$username:$password")
        )
    );
    $context = stream_context_create($opts);
    $response = file_get_contents($url, false, $context);
    return json_decode($response, true);
}

$pid = getenv('KK_PERSON_ID') ?: 'testuser';
print_r(get_kellokortti_person($pid));
?>
